tests/Doctrine/Tests/ORM/Functional/EnumTest.php

```diff
@@ -115,6 +118,59 @@ public function testEnumArrayHydration(): void
         self::assertEqualsCanonicalizing([Unit::Gram, Unit::Meter], $result[0]['supportedUnits']);
     }
 
+    public function testEnumInDtoHydration(): void
+    {
+        $this->setUpEntitySchema([Card::class, CardWithNullable::class]);
+
+        $card       = new Card();
+        $card->suit = Suit::Clubs;
+
+        $cardWithNullable       = new CardWithNullable();
+        $cardWithNullable->suit = null;
+
+        $this->_em->persist($card);
+        $this->_em->persist($cardWithNullable);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $result = $this->_em->createQueryBuilder()
+            ->from(Card::class, 'c')
+            ->select(new Func('NEW ' . DtoWithEnum::class, ['c.suit']))
+            ->getQuery()
+            ->getResult();
+
+        $this->assertInstanceOf(Suit::class, $result[0]->suit);
+        $this->assertEquals(Suit::Clubs, $result[0]->suit);
+
+        $result = $this->_em->createQueryBuilder()
+            ->from(CardWithNullable::class, 'c')
+            ->select(new Func('NEW ' . DtoWithEnum::class, ['c.suit']))
+            ->getQuery()
+            ->getResult();
+
+        $this->assertNull($result[0]->suit);
```

Comment:
These are actually two separate test cases. Please split them up.

Reply:
Done. 👍 

As it was inspired by a test case above, I modified it as well.